docs/source/format/DissociatedIPC.rst

```diff
@@ -0,0 +1,335 @@
+
+       As usage among different transports and libraries grows, common patterns will emerge
+       that will allow for a generic, but efficient, way to handle backpressure across
+       different use cases.
+
+       .. note::
+          While the protocol itself is transport agnostic, the current usage and examples 
+          only have been tested using UCX and libfabric transports so far, but that's all.
+
+
+Protocol Description
+====================
+
+A reference example implementation utilizing `libcudf`_ and `ucx`_ can be found at 
+https://github.com/zeroshade/cudf-flight-ucx.
+
+Requirements
+------------
+
+A transport implementing this protocol **MUST** provide two pieces of functionality:
+
+* Message sending
+  - Delimited messages (like gRPC) as opposed to non-delimited streams (like plain TCP 
+    without further framing).
+  - Alternatively, a framing mechanism like the `encapsulated message format <ipc-message-format>`
+    for the IPC protocol can be used while leaving out the body bytes.
+* Tagged message sending
+  - Sending a message that has an attached little-endian, unsigned 64-bit integral tag
+    for control flow. A tag like this allows control flow to operate on a message whose body
+    is on a non-CPU device without requiring the message itself to get copied off of the device.
+
+URI Specification
+-----------------
+
+When providing a URI to a consumer to contact for use with this protocol (such as via 
+the `Location URI for Flight <flight-location-uris>`_), the URI should specify a scheme
+like *ucx://* or *fabric://*, that is easily identifiable. In addition, the URI should
+encode the following URI query parameters:
+
+.. note::
+    As this protocol matures, this document will get updated with commonly recognized
+    transport schemes that get used with it.
+
+* ``want_data`` - **REQUIRED** - uint64 integer value
+  - This value should be used to tag an initial message to the server to initiate a
+    data transfer. The body of the initiating message should be an opaque binary identifier
+    of the data stream being requested (like the ``Ticket`` in the Flight RPC protocol)
+* ``free_data`` - **OPTIONAL** - uint64 integer value
+  - If the server might send messages using offsets / addresses for remote memory accessing
+    or shared memory locations, the URI should include this parameter. This value is used to
+    tag messages sent from the client to the data server, containing specific offsets / addresses
+    which were provided that are no longer required by the client (i.e. any operations that
+    directly reference those memory locations, such as copying the remote data into local memory,
+    have been completed).
+* ``remote_handle`` - **OPTIONAL** - base64-encoded string
+  - When working with shared memory or remote memory, this value indicates any required
+    handle or identifier that is necessary for accessing the memory.
+    + Using UCX, this would be an *rkey* value
```

Comment:
Uh... if this parameter is transport-specific, then should it really be documented here? Also, does it need to be a query parameter? What does an UCX URI look like?

Reply:
the interpretation of what the parameter means is transport-specific. The documentation here is essentially stating examples of how you would use this parameter for certain transports. 

There is no real standard for a "UCX URI". You only use the host/IP to connect to a server, the address can also be provided via a serialized opaque Worker Address object.